tests/test_conda_env.py

```diff
@@ -1,459 +1,315 @@
-    assert "conda" in call.args[0]
-    assert "create" == call.args[1]
-    assert "--yes" == call.args[2]
-    assert "-p" == call.args[3]
-    assert venv.path == call.args[4]
-    assert call.args[5].startswith("python=")
+def test_conda_create_with_package_install(tox_project, mock_conda_env_runner):
+    ini = """
+    [testenv:py123]
+    """
+    proj = tox_project({"tox.ini": ini})
 
+    outcome = proj.run("-e", "py123")
+    outcome.assert_success()
 
-def create_test_env(config, mocksession, envname):
+    executed_shell_commands = mock_conda_env_runner
+    assert len(executed_shell_commands) == 3
+    conda_create_command = executed_shell_commands[1]
+    pip_install_command = executed_shell_commands[2]
 
-    venv = VirtualEnv(config.envconfigs[envname])
-    with mocksession.newaction(venv.name, "getenv") as action:
-        tox_testenv_create(action=action, venv=venv)
-    pcalls = mocksession._pcalls
-    assert len(pcalls) >= 1
-    pcalls[:] = []
+    assert fnmatch(
+        conda_create_command,
+        f"*conda create -p {str(proj.path / '.tox' / 'py123')} python=* --yes --quiet*",
+    )
+    assert fnmatch(
+        pip_install_command,
+        (
+            f"*conda run -p {str(proj.path / '.tox' / 'py123')} --live-stream"
+            " python -I -m pip install*"
+        ),
+    )
 
-    return venv, action, pcalls
 
+def test_conda_create_with_name(tox_project, mock_conda_env_runner):
+    ini = """
+    [testenv:py123]
+    skip_install = True
+    conda_name = myenv
+    """
+    proj = tox_project({"tox.ini": ini})
 
-def test_install_deps_no_conda(newconfig, mocksession, monkeypatch):
-    """Test installation using conda when no conda_deps are given"""
-    # No longer remove the temporary script, so we can check its contents.
-    monkeypatch.delattr(PopenInActivatedEnv, "__del__", raising=False)
+    outcome = proj.run("-e", "py123")
+    outcome.assert_success()
 
+    executed_shell_commands = mock_conda_env_runner
+    assert len(executed_shell_commands) == 2
+    assert fnmatch(executed_shell_commands[1], "*conda create -n myenv python=* --yes --quiet*")
+
+
+def test_install_deps_no_conda(tox_project, mock_conda_env_runner):
     env_name = "py123"
-    config = newconfig(
-        [],
-        """
-        [testenv:{}]
-        deps=
+    ini = f"""
+        [testenv:{env_name}]
+        skip_install = True
+        deps =
             numpy
             -r requirements.txt
             astropy
-    """.format(
-            env_name
+    """
+    proj = tox_project({"tox.ini": ini})
+    (proj.path / "requirements.txt").touch()
+
+    outcome = proj.run("-e", "py123")
+    outcome.assert_success()
+
+    executed_shell_commands = mock_conda_env_runner
+    assert len(executed_shell_commands) == 3
+
+    pip_install_command = executed_shell_commands[2]
+    assert fnmatch(
+        pip_install_command,
+        (
+            f"*conda run -p {str(proj.path / '.tox' / env_name)} --live-stream"
```

Comment:
@gaborbernat Let me ask your advise here. All the tests are based on the fixtures (`tox_project`) that are copy (with minor modification) from the `tox.pytest` module. It works properly, when I call `pytest` directly, all tests are passing. However, when I use `tox`, it does not work well, `proj.path` point to the source root folder instead of the temporary directory. I am leaning towards running `tox-conda` test using `pytest` for now, but I do not particularly like this, I want to keep `tox` here. Is there any solution for this?

Reply:
Would have to check why the path differs.

Reply:
It is actually more, `tox_project` creates the tox environment in the `.tox` folder of the source folder.